components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4130,4 +4140,64 @@ protected void checkAccessControlPermission(APIIdentifier identifier) throws API
         return updatedWSDLContent;
     }
 
+    /**
+     * This method is used to get keys of custom attributes, configured by user
+     *
+     * @param userId user name of logged in user
+     * @return Array of JSONObject, contains keys of attributes
+     * @throws APIManagementException
+     */
+    public JSONArray getAppAttributesFromConfig(String userId) throws APIManagementException {
+
+        String tenantDomain = MultitenantUtils.getTenantDomain(userId);
+        int tenantId = 0;
+        try {
+            tenantId = getTenantId(tenantDomain);
+        } catch (UserStoreException e) {
+            handleException("Error in getting tenantId of " + tenantDomain, e);
+        }
+        JSONArray applicationAttributes = null;
+        JSONObject applicationConfig = APIUtil.getAppAttributeKeysFromRegistry(tenantId);
+        try {
+            if (applicationConfig != null) {
+                applicationAttributes = (JSONArray) applicationConfig.get(APIConstants.ApplicationAttributes.ATTRIBUTES);
+            } else {
+                APIManagerConfiguration configuration = getAPIManagerConfiguration();
+                applicationAttributes = configuration.getApplicationAttributes();
+            }
+        } catch (NullPointerException e){
```

Comment:
Any special need to catch an NPE? Shouldn't we avoid that in the first place?

Reply:
Yes, it can be removed. Resolved in #6098